wezterm-input-types/src/lib.rs

```diff
@@ -1800,30 +1812,31 @@ impl KeyEvent {
                 format!("\x1b[{c};{modifiers}{event_type}~")
             }
             Char(shifted_key) => {
-                let mut use_legacy = false;
                 let shifted_key = if *shifted_key == '\x08' {
                     // Backspace is really VERASE -> ASCII DEL
                     '\x7f'
                 } else {
                     *shifted_key
                 };
 
-                if !flags.contains(KittyKeyboardFlags::REPORT_ALTERNATE_KEYS)
+                let use_legacy = !flags.contains(KittyKeyboardFlags::REPORT_ALTERNATE_KEYS)
                     && event_type.is_empty()
                     && is_legacy_key
                     && !(flags.contains(KittyKeyboardFlags::DISAMBIGUATE_ESCAPE_CODES)
                         && (self.modifiers.contains(Modifiers::CTRL)
                             || self.modifiers.contains(Modifiers::ALT)))
-                {
-                    use_legacy = true;
-                }
+                    && !self.modifiers.contains(Modifiers::SUPER)
+                    && !self.modifiers.contains(Modifiers::META)
+                    && !self.modifiers.contains(Modifiers::HYPER);
```

Comment:
This is probably a bit more elegant if rephrased as something like:

```rust
!self.modifiers.intersects(Modifiers::SUPER|Modifiers::META|Modifiers::HYPER)
```

Reply:
Ah jeah, I was looking for that thing, something like `modifiers.any` but didnt exist and was too dump to construct this. Much better!